src/internal/connector/exchange/exchange_data_collection_test.go

```diff
@@ -160,3 +166,47 @@ func (suite *ExchangeDataCollectionSuite) TestNewCollection_state() {
 		})
 	}
 }
+
+func (suite *ExchangeDataCollectionSuite) TestGetItemWithRetries() {
+	table := []struct {
+		name      string
+		items     mockItemer
+		expectErr func(*testing.T, error)
+	}{
+		{
+			name:  "happy",
+			items: mockItemer{},
+			expectErr: func(t *testing.T, err error) {
+				assert.NoError(t, err)
+			},
+		},
+		{
+			name:  "an error",
```

Comment:
this test is good to have, but it would also be good to have a test where we actually count the number of times the retry is called. That will help ensure the retry is actually working as expected. It adds a little more test complexity but it can help track down issues as well

One of the recent OneDrive PRs I have does a test to check the number of retries and it programmatically generates the mock results by reusing the variable containing the number of retries https://github.com/alcionai/corso/pull/2306/files#diff-a63d5466edbe22450c15d02fa6f9dd0cf04785314552c509e3c8c4d334e85978R120

Reply:
oh yeah, not a bad idea.  Added that.